packages/sanes-chrome-extension/src/extension/backgroundscript.ts

```diff
@@ -22,6 +22,26 @@ async function handleMessageToBackground(
         console.log('Received message from: ' + sender.tab.id);
       }
       return { msg: 'Hello mate' };
+    case MessageToBackgroundAction.GetPersona: {
+      if (sender.id !== chrome.runtime.id) {
+        return 'Sender is not allowed to perform this action';
+      }
+      let persona: Persona | undefined;
```

Comment:
I would do something like: 
```
try {
   const persona = PersonaManager.get(); <--- remember if persona does not exist it throws an exception (this functionality should be tested for assuring the exception is raised)
   return {
       mnemonic: persona.mnemonic,
       txs: await persona.getTxs(),
       accounts: await persona.getAccounts(),
    }
} catch () {
    return null;
}
```

Reply:
The important difference here is that I do not want to catch exceptions from `await persona.getTxs()` and `await persona.getAccounts()`. If I catch them, errors are interpreted as _no persona_ and not detected.

Reply:
What happens if `getAccounts`  throws an Error? where is it caught?

What about...
```
try {
   const persona = PersonaManager.get(); <--- remember if persona does not exist it throws an exception (this functionality should be tested for assuring the exception is raised)
   return {
       mnemonic: persona.mnemonic,
       txs: await persona.getTxs(),
       accounts: await persona.getAccounts(),
    }
} catch (err) {
    console.log(err); <--- maybe here you can see where the error comes from
    return null;
}
```

Or what do you want to do with the error if it comes from `getAccounts`? I am curious.

Reply:
Just for giving more context, the main reason because I wrote this comment, was the difficulty to read the code.

Reply:
> What happens if `getAccounts` throws an Error? where is it caught?

A primose rejection in handleMessageToBackground.

> the main reason because I wrote this comment, was the difficulty to read the code.

Yeah, there is a much simpler way to improve readability: Remove the obsolete `PersonaManager` and store the persona instance in the BS directly. I can make this as part of this PR if you like.

Reply:
> A primose rejection in handleMessageToBackground.

Sounds good.

> store the persona instance in the BS directly

No, please, the BS will change in the incoming PRs, once it is stabilised we can think about it, but leave it as it is now.



👌